core/src/main/java/org/apache/iceberg/actions/BaseExpireSnapshotsActionResult.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+public class BaseExpireSnapshotsActionResult implements ExpireSnapshots.Result {
+
+  private final long deletedDataFilesCount;
```

Comment:
Seems we don't consider the format v2 in this patch, right ?  That's OK if true because we could support v2 in a separate issue.  For the RewriteDataFilesAction, now @chenjunjiedada and I have prepared few PRs to support format v2's Rewrite actions.  

1. https://github.com/apache/iceberg/pull/2303 
2. https://github.com/apache/iceberg/pull/2216

Reply:
Yep, you are correct. I keep the scope of this PR to the existing logic. 

@RussellSpitzer is also working on redesigning the rewrite data files action to also support sort-based compactions and per partition compactions. Shall 4 of us meet next week to discuss this?

Reply:
It would be easy to extend the interface and this implementation to include other counts once we support them.